src/fb.h

```diff
@@ -5,13 +5,30 @@
 
 #include "types.h"
 
-extern u32 *fb;
-extern int fb_s, fb_w, fb_h;
+typedef struct {
+    u32 *ptr;   /* pointer to the start of the framebuffer */
+    u32 stride; /* framebuffer stride divided by four (i.e. stride in pixels) */
+    u32 depth;  /* framebuffer depth (i.e. bits per pixel) */
+    u32 width;  /* width of the framebuffer in pixels */
+    u32 height; /* height of the framebuffer in pixels */
+} fb_t;
+
+typedef struct {
+    u8 r;
+    u8 g;
+    u8 b;
+} rgb_t;
+
+extern fb_t fb;
```

Comment:
Same nitpick as fb_t declaration in fb.c, we don't use this anywhere else so it really doesn't have to be a global like that.

Reply:
If we make fb private we might as well make fb_blit etc. private because you can't tell the width and height of the fb anymore. this could be cleaned up in a separate change but I'll leave it the way it is right now since there are already a lot of changes at this point.

Reply:
Fair point. I was mostly going off how nothing else _currently_ uses it and wasn't thinking about the fact that it was meant to be part of the public interface for potential future use. In that case it's probably fine.